File: Practica1/Ejercicio4/src/GestorFiltros.java
public class GestorFiltros {
    private CadenaFiltros cadenaFiltros;

    public GestorFiltros(Objetivo salpicadero){
        cadenaFiltros = new CadenaFiltros(salpicadero);
    }

    public void addFiltro( Filtro filtro){
        cadenaFiltros.addFiltro(filtro);
    }

    public double ejecutar(double revoluciones, EstadoMotor estadoMotor){
        return cadenaFiltros.ejecutar(revoluciones, estadoMotor);
    }
}

File: Practica3/lib/lib/gestorFiltros.dart
import 'package:ejercicio3/producto.dart';
import 'package:ejercicio3/filtro.dart';
import 'package:ejercicio3/filtroStruct.dart';
import 'package:ejercicio3/cadenaFiltros.dart';

class GestorFiltros {
  late CadenaFiltros cadenaFiltros;

  GestorFiltros(List<Producto> productos) {
    cadenaFiltros = CadenaFiltros(productos);
  }

  void addFiltro(Filtro filtro) {
    cadenaFiltros.addFiltro(filtro);
  }

  List<Producto> filtrar(FiltroStruct filtrosAplicar) {
    return cadenaFiltros.filtrar(filtrosAplicar);
  }
}
